### PR TITLE
Dont call the same CouchDB view twice; support status+type+requestor ReqMgr2 REST API

### LINKED ISSUE #9419: reqmgr2 api gives internal server error for relval workflows
**Impact of the bug**
Batchor module is affected in unified which causes the relvals to stop working. 

**Describe the bug**
http://cmsweb.cern.ch/reqmgr2/data/request?status=assignment-approved&request_type=TaskChain&requestor=kaura&detail=true is giving 501 internal server error

**How to reproduce it**
http://cmsweb.cern.ch/reqmgr2/data/request?status=assignment-approved&request_type=TaskChain&requestor=kaura&detail=true

**Expected behavior**
should return a proper json

**Additional context and error message**

### PR DESCRIPTION
**Pull request by amaltaro**
Fixes #9419 

#### Status
not-tested

#### Description
Summary of changes is:
* implemented a new CouchDB view named `bystatusandtypeandrequestor`. It's suppose to yield: RequestStatus, RequestType and Requestor
* fixed the `request` ReqMgr2 REST API, such that it doesn't call the same CouchDB view twice (like the `bystatus`)
* overall reorganization of how we call the couchdb views to avoid overloading the ReqMgr2 backend.

#### Is it backward compatible (if not, which system it affects?)
Yes, besides the new couchdb view.

#### Related PRs
none

#### External dependencies / deployment changes
none

### CODE REVIEW

**Comment by amaltaro** [comment]
test this please

**Comment by amaltaro** [comment]
Tests are looking good now. @todor-ivanov can you please review this patch? Thanks

File: src/python/WMCore/ReqMgr/Service/Request.py [hunk 1]
```diff
@@ -285,68 +290,88 @@ def get(self, **kwargs):
...
+            status = ACTIVE_STATUS
         if detail in (False, "false", "False", "FALSE"):
             option = {"include_docs": False}
         else:
             option = {"include_docs": True}
-        # eventhing should be stale view. this only needs for test
-        _nostale = kwargs.get("_nostale", False)
-        if _nostale:
+        # everything should be stale view. this only needs for test
+        if nostale:
```
Comment (todor-ivanov) [inline src/python/WMCore/ReqMgr/Service/Request.py]: Are we sure we want to introduce a discrepancy between the key name in the document and the key name we use in the code

**Review by todor-ivanov — APPROVED** [review]
Besides the one question regarding the change of a variable name (see the comments in line), I have no other remarks. All the rest looks fine to me.

File: src/python/WMCore/ReqMgr/Service/Request.py (hunk 1 above)
Comment (amaltaro) [reply src/python/WMCore/ReqMgr/Service/Request.py]: That's really just a flag, not used for anything else.

**Comment by amaltaro** [comment]
test this please

**Pull request merged**